wasmbinding/message_plugin.go

```diff
@@ -62,11 +67,48 @@ func (m *CustomMessenger) DispatchMsg(ctx sdk.Context, contractAddr sdk.AccAddre
 		if contractMsg.RemoveInterchainQuery != nil {
 			return m.removeInterchainQuery(ctx, contractAddr, contractMsg.RemoveInterchainQuery)
 		}
+		if contractMsg.IBCTransferMsg != nil {
+			return m.ibcTransfer(ctx, contractAddr, *contractMsg.IBCTransferMsg)
+		}
 	}
 
 	return m.Wrapped.DispatchMsg(ctx, contractAddr, contractIBCPortID, msg)
 }
 
+func (m *CustomMessenger) ibcTransfer(ctx sdk.Context, contractAddr sdk.AccAddress, ibcTransferMsg transferwrappertypes.MsgTransfer) ([]sdk.Event, [][]byte, error) {
+	ibcTransferMsg.Sender = contractAddr.String()
```

Comment:
it contains a value indeed, but we can't trust it, because the contract sets it. And this means that a malicious contract can set `.Sender` to any address and executing transfer from any address, which is not right. This line fixes the problem